src/qibocal/auto/task.py

```diff
@@ -74,28 +67,34 @@ def priority(self):
 
     @property
     def parameters(self):
-        return Parameters.load(self.action.parameters)
+        return self.operation.parameters_type.load(self.action.parameters)
 
     def datapath(self, base_dir: Path):
-        return base_dir / f"{self.id}_{self.iteration}" / DATAFILE
+        path = base_dir / "data" / f"{self.id}_{self.iteration}"
+        os.makedirs(path)
+        return path
 
     def data(self, base_dir) -> Optional[Data]:
-        if not self.datapath(base_dir).is_file():
+        if not self.datapath(Path(base_dir)).is_file():
```

Comment:
```suggestion
        if not self.datapath(base_dir).is_file():
```

Just ask for a `Path` in the type hint, at least we avoid calling `Path()` in tons of places (we should convert them just in the interface, and then assume they are `Path` objects).